### ISSUE TITLE
Segfault with hpfeeds ihandler present

### ISSUE THREAD

**Issue by d1str0** [DinoTools/dionaea#144]
##### ISSUE TYPE
 - Bug Report

##### DIONAEA VERSION

02492e2

##### CONFIGURATION

default hpfeeds ihandler with config variables changed to my MHN server and dynip set to "http://icanhazip.com/"
##### OS / ENVIRONMENT

Docker using ubuntu:16.04 base image, or debian:stretch-slim base image.

Dockerfile: https://gist.github.com/8e9f2bf9757b91a8617b54e92319033b

##### SUMMARY

I get a segfault after about 1 min of running, only when hpfeeds ihandler is present.

I saw the previous issue (#115) talking about a 60 second or so segfault and it mentioned a previously stable commit (be43edc). When I tested this one I also get a segfault after about 60 seconds (with hpfeeds ihandler).
##### STEPS TO REPRODUCE

Using above dockerfile, docker build. Then with that image,
```
docker run -p 21:21 -p 42:42 -p 69:69/udp -p 8080:80 -p 135:135 -p 443:443 -p 445:445 -p 1433:1433 -p 1723:1723 -p 1883:1883 -p 1900:1900/udp -p 3306:3306 -p 5060:5060 -p 5060:5060/udp -p 5061:5061 -p 11211:11211 -v ~/dionaea:/opt/dionaea/etc/dionaea  image_name:latest
```

And inside ~/dionaea I have dionaea.cfg: https://gist.github.com/8a27ebaff332fafe560315c6e56662e2

And inside ~/dionaea/ihandlers I have hpfeeds.yaml: https://gist.github.com/f74141667073bfde5744dcc3cc5bec56
(with proper config values)

##### EXPECTED RESULTS

I expect it to run and talk back to my MHN server.

##### ACTUAL RESULTS

It does not manage to talk to my MHN server, also it segfaults after ~60 seconds.

log: https://gist.github.com/c2d0ca1ab1868b0d98b1a66b75f59891
(this log is incomplete, working on getting a new one)

Segfault message:
```
This is the end.                                   
This software just had a segmentation fault.       
The bug you encountered may even be exploitable.   
If you want to assist in fixing the bug, please create a new issue on GitHub and append the backtrace.                                                                                                      
You can create better backtraces with gdb, for more information visit https://github.com/DinoTools/dionaea                                                                                                  
Once you read this message, your tty may be broken, simply type reset, so it will come to life again  

./dionaea(sigsegv_backtrace_cb+0x35)[0x41b0c2]     
/lib/x86_64-linux-gnu/libc.so.6(+0x354b0)[0x7f21069af4b0]                                             
/lib/x86_64-linux-gnu/libc.so.6(inet_pton+0x2b)[0x7f2106aa022b]                                       
./dionaea(parse_addr+0x53)[0x40cb31]               
./dionaea(connection_reconnect_timeout_cb+0x115)[0x4105ab]                                            
/usr/lib/x86_64-linux-gnu/libev.so.4(ev_invoke_pending+0x53)[0x7f2107d32d73]                          
/usr/lib/x86_64-linux-gnu/libev.so.4(ev_run+0x9ee)[0x7f2107d363de]                                    
./dionaea(main+0x13ff)[0x40ba96]                   
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7f210699a830]                               
./dionaea(_start+0x29)[0x409889] 
```

#### Additional Note
When I run the dinotools/dionaea-docker 0.6.0 image with my hpfeeds.yaml and same config file, I *do* get results back to my MHN server, however it still segfaults every 60 seconds or so, but in this case is run supervisor and restarts every time it segfaults.

**Comment by d1str0** [comment]
This might be related to #138

**Comment by iamalsaher** [comment]
So did you figure out anything, I am facing the same problem on my Raspberry Pi but if I install on Ubuntu server, it works perfectly.....

**Comment by phibos** [comment]
The bug should have been fixed with PR #217, has been merged into the master branch and will be released in version 0.9.0.

The hpfeeds ihandler will also get some improvements feel free to have a look at #215.

Please reopen the issue or create a new one if you have still issues.
